src/mm-rating/mm-rating.js

```diff
@@ -0,0 +1,113 @@
+( function (scope) {
+    scope.Rating = Polymer({
+        is: 'mm-rating',
+        hovered: false,
+        properties: {
+            icon: {
+                type: String,
+                value: 'star'
+            },
+            customIcon: {
+                type: String,
+                value: null
+            },
+            iconWidth: {
+                type: Number,
+                value: 20
+            },
+            iconHeight: {
+                type: Number,
+                value: 20
+            },
+            rating: {
+                type: Number,
+                value: 0
+            },
+            displayTotals: {
+                type: Boolean,
+                value: false
+            },
+            displayGreetings: {
+                type: Boolean,
+                value: false
+            },
+            readonly: {
+                type: Boolean,
+                value: false
+            },
+            values: {
+                type: Object,
+                value: {0: 'Poor', 1: 'Need Improvement', 2: 'Reasonable', 3: 'Good', 4: 'Excellent'}
+            }
+        },
+
+        ready: function() {
+            this._defaultNoOfIcons = [];
+            if(typeof this.values === 'string'){
+                var greetObj = eval("("+ this.values +")");
+                this._noOfIcons = Object.keys(greetObj).length;
+            } else if(typeof this.values === 'object'){
+                this._noOfIcons = Object.keys(this.values).length;
+            }
+            for (var i = this._noOfIcons - 1; i >= 0; i--) {
+                this._defaultNoOfIcons.push(i+1);
+            }
+        },
+        isCustomIcon: function() {
+            return this.customIcon;
+        },
+        isActive: function(index) {
+            if ((index - this._noOfIcons) * -1 == this.rating) {
+                return "active";
+            }
+        },
+        isDisplayTotals: function(showtotals) {
+            return showtotals;
+        },
+        isDisplayGreetings: function(showgreetings) {
+            return showgreetings;
+        },
+        calculateRating: function(e) {
+            deep = Polymer.dom(this.root);
+            if(!this.readonly){
+                var index = e.model.index;
+                var indexOld = (this.rating * -1) + this._noOfIcons;
+                this.rating = (index - this._noOfIcons) * -1;
+                if (indexOld < this._noOfIcons) {
+                    deep.querySelector('[data-index="'+ indexOld +'"]').classList.remove("active");
+                }
+                deep.querySelector('[data-index="'+ index +'"]').classList.add("active");
```

Comment:
we have a behavior to handle this situation:

https://github.com/MediaMath/strand/blob/master/src/shared/behaviors/stylable.html#L22

that allows for binding the classBlock through to your data so you don't have to manually update this.